TramsDataApi.Test/Utils/Generators.cs

```diff
@@ -0,0 +1,96 @@
+                Ks4Attainment8Score = Decimal.Parse("0.0"),
+                Ks4Progress8Score = Decimal.Parse("0.0"),
+                Ks4DisadvProgress8Score = Decimal.Parse("0.0"),
+                Ks4AvgAchieveBasics3Years = Decimal.Parse("0.0"),
+                Ks4CoastingFlag = randomGenerator.Boolean(),
+                Ks2ExpStandardsRwm = Decimal.Parse("0.0"),
+                Ks2Reading = Decimal.Parse("0.0"),
+                Ks2Maths = Decimal.Parse("0.0"),
+                Ks2Writing = Decimal.Parse("0.0"),
+                Ks2DisadvReading = Decimal.Parse("0.0"),
+                Ks2DisadvMaths = Decimal.Parse("0.0"),
+                Ks2DisadvWriting = Decimal.Parse("0.0"),
+                Ks2CoastingFlag = randomGenerator.Boolean(),
+                AbsencesOverall = Decimal.Parse("0.0"),
+                AbsencesPa = Decimal.Parse("0.0"),
+                AbsencesUnauthorised = Decimal.Parse("0.0"),
+                PsdFlag = randomGenerator.Char().ToString(),
+                PredecessorUrn = randomGenerator.NextString(8, 8),
+                PredecessorName = randomGenerator.NextString(1, 100),
+                OfstedSource = randomGenerator.Int(),
+                Ks4Source = randomGenerator.Int(),
+                Ks2Source = randomGenerator.Int(),
+                AbsSource = randomGenerator.Int(),
+                PsdSource = randomGenerator.Int(),
+                FlagOfstedEverInadequate = randomGenerator.Int(),
+                FlagOfstedLastTwoRi = randomGenerator.Int(),
+                FlagHtchangesTotal = randomGenerator.Int(),
+                FlagHtchangesLastYear = randomGenerator.Int(),
+                FlagKs4Attainment8 = randomGenerator.Int(),
+                FlagKs4Progress8Score = randomGenerator.Int(),
+                FlagKs4AvgAchieveBasics3Years = randomGenerator.Int(),
+                FlagKs4DisadvProgress8Score = randomGenerator.Int(),
+                FlagKs4CoastingFlag = randomGenerator.Int(),
+                FlagKs2ExpStandardsRwm = randomGenerator.Int(),
+                FlagKs2CombinedProgress = randomGenerator.Int(),
+                FlagKs2CombinedDisadvantagedProgress = randomGenerator.Int(),
+                FlagKs2CoastingFlag = randomGenerator.Int(),
+                FlagAbsencesUnauthorised = randomGenerator.Int(),
+                FlagAbsencesOverallPa = randomGenerator.Int(),
+                IsNa = randomGenerator.Boolean(),
+                DateOfLastFullOrShortInspection = randomGenerator.DateTime(),
+                ProbabilityOfDeclining = randomGenerator.Float(),
+                ProbabilityOfStayingTheSame = randomGenerator.Float(),
+                ProbabilityOfImproving = randomGenerator.Float(),
+                PredictedChangeInProgress8Score = randomGenerator.NextString(1, 23),
+                PredictedChanceOfChangeOccuring = randomGenerator.Float(),
+                TotalNumberOfRisks = randomGenerator.Int(),
+                TotalRiskScore = Decimal.Parse("0.0"),
```

Comment:
Can you tell me a bit more about why this is a decimal/not randomised? I'm just curious :D

Reply:
I did a quick writeup on the commit - https://github.com/DFE-Digital/trams-data-api/pull/14/commits/32c7ced909d48ab2d8fb127c519e0ee8b0c5ee07 

Basically because the database has rules on the floating-point precision of some of the fields,  I was finding that randomly generating the values was causing overflow errors - even when I specifically had restrictions on the values. It caused some headscratchingly-flaky tests, so to eliminate them I decided to be specific on the values. We get reasonable entropy from the other random values, so there's not much risk of impacting the triangulation aspect of the tests